tools/wptrunner/wptrunner/browsers/safari.py

```diff
@@ -60,6 +61,79 @@ def env_options():
     return {}
 
 
+def run_info_extras(**kwargs):
+    webdriver_binary = kwargs["webdriver_binary"]
+    rv = {}
+
+    safari_bundle, safari_info = get_safari_info(webdriver_binary)
+
+    if safari_info is not None:
+        assert safari_bundle is not None  # if safari_info is not None, this can't be
+        _, webkit_info = get_webkit_info(safari_bundle)
+        if webkit_info is None:
+            webkit_info = {}
+    else:
+        safari_info = {}
+        webkit_info = {}
+
+    rv["browser_marketing_version"] = safari_info.get("CFBundleShortVersionString")
+    rv["browser_bundle_version"] = safari_info.get("CFBundleVersion")
+    rv["browser_webkit_bundle_version"] = webkit_info.get("CFBundleVersion")
+
+    with open("/System/Library/CoreServices/SystemVersion.plist", "rb") as fp:
```

Comment:
I assume this isn't fallible under reasonable circumstances?

Reply:
I'm just taking the view of "if you start modifying system files" (which requires disabling SIP _anyway_) then all bets are out about… everything. Strongly suspect you'll run into OS-provided things breaking!